packages/@n8n/ai-workflow-builder.ee/src/tools/prompts/main-agent.prompt.ts

```diff
@@ -178,6 +178,32 @@ Common failures from relying on defaults:
 ALWAYS check node details obtained in Analysis Phase and configure accordingly. Defaults are NOT your friend - they are traps that cause workflows to fail at runtime.
 </node_defaults_warning>
 
+<workflow_configuration_node>
+CRITICAL: Always include a Workflow Configuration node at the start of every workflow.
+
+The Workflow Configuration node (n8n-nodes-base.set) is a mandatory node that should be placed immediately after the trigger node and before all other processing nodes.
+This node centralizes workflow-wide settings and parameters that other nodes can reference throughout the execution with expressions.
+
+Placement rules:
+- ALWAYS add between trigger and first processing node
+- Connect: Trigger → Workflow Configuration → First processing node
+- This creates a single source of truth for workflow parameters
+
+Configuration approach:
+- Include URLs, thresholds, string constants and any reusable values
+- Other nodes reference these via expressions: {{ $('Workflow Configuration').item.json.variableName }}
```

Comment:
`.first()` might be more reliable here. 

for example

start -> workflow config -> node returns multiple items, but breaks paired item -> references config